runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/HttpBody.kt

```diff
@@ -104,21 +104,30 @@ public fun String.toHttpBody(): HttpBody = encodeToByteArray().toHttpBody()
  */
 @InternalApi
 public fun ByteStream.toHttpBody(): HttpBody = when (val byteStream = this) {
-    is ByteStream.Buffer -> object : HttpBody.Bytes() {
-        override val contentLength: Long? = byteStream.contentLength
-        override fun bytes(): ByteArray = byteStream.bytes()
-    }
+    is ByteStream.Buffer -> ByteStreamBufferHttpBody(byteStream)
+    is ByteStream.ChannelStream -> ByteStreamChannelHttpBody(byteStream)
+    is ByteStream.SourceStream -> ByteStreamSourceHttpBody(byteStream)
+}
 
-    is ByteStream.ChannelStream -> object : HttpBody.ChannelContent() {
-        override val contentLength: Long? = byteStream.contentLength
-        override val isOneShot: Boolean = byteStream.isOneShot
-        override fun readFrom(): SdkByteReadChannel = byteStream.readFrom()
-    }
-    is ByteStream.SourceStream -> object : HttpBody.SourceContent() {
-        override val contentLength: Long? = byteStream.contentLength
-        override val isOneShot: Boolean = byteStream.isOneShot
-        override fun readFrom(): SdkSource = byteStream.readFrom()
-    }
+private interface ByteStreamHttpBody {
+    val stream: ByteStream
+}
+
+private class ByteStreamBufferHttpBody(override val stream: ByteStream.Buffer) : ByteStreamHttpBody, HttpBody.Bytes() {
+    override val contentLength: Long? = stream.contentLength
+    override fun bytes(): ByteArray = stream.bytes()
+}
+
+private class ByteStreamChannelHttpBody(override val stream: ByteStream.ChannelStream) : ByteStreamHttpBody, HttpBody.ChannelContent() {
+    override val contentLength: Long? = stream.contentLength
+    override val isOneShot: Boolean = stream.isOneShot
+    override fun readFrom(): SdkByteReadChannel = stream.readFrom()
+}
+
+private class ByteStreamSourceHttpBody(override val stream: ByteStream.SourceStream) : ByteStreamHttpBody, HttpBody.SourceContent() {
+    override val contentLength: Long? = stream.contentLength
+    override val isOneShot: Boolean = stream.isOneShot
+    override fun readFrom(): SdkSource = stream.readFrom()
 }
```

Comment:
question: what is the purpose of introducing these new `ByteStreamHttpBody` implementations